src/coreComponents/functions/MultivariableTableFunctionKernels.hpp

```diff
@@ -205,9 +217,60 @@ class MultivariableTableFunctionStaticKernel
     return axisIntervalIndex;
   }
 
+  /**
+   * @brief interpolate all operators values at a given point
+   * The algoritm is based on http://dx.doi.org/10.1090/S0025-5718-1988-0917826-0
+   *
+   * @param[in] axisCoordinates coordinates of a point
+   * @param[in] hypercubeData data of target hypercube
+   * @param[in] axisLows array of left coordinates of target axis intervals
+   * @param[in] axisStepInvs array of inversions of axis steps
+   * @param[out] values interpolated operator values
+   */
+  template< typename IN_ARRAY, typename OUT_ARRAY >
+  GEOSX_HOST_DEVICE
+  inline
+  void
+  interpolatePoint( IN_ARRAY const & axisCoordinates,
+                    real64 const * const hypercubeData,
```

Comment:
Yes, I agree with you that we should try to pass a `arraySlice2d` here. 

The underlying array `m_hyprecubeData` lives in the table function class, here. I think it could be a `array3d` of dimensions `numTableHypercubes`,`numVerts`, and `numOps`, but currently it is just stored as an `array1d` of size `numTableHypercubes x numVerts x numOps`. Then, we would call `interpolatePoint` as:

```
interpolatePoint( coordinates,                                                                                                                                                       
                             m_hypercubeIndex[hypercubeIndex],                                                                                                                                
                             &axisLows[0],                                                                                                                                                      
                             &m_axisStepInvs[0],                                                                                                                                                
                             values );  
```
instead of
```
interpolatePoint( coordinates,                                                                                                                                                       
                      getHypercubeData( hypercubeIndex ),                                                                                                                                
                      &axisLows[0],                                                                                                                                                      
                      &m_axisStepInvs[0],                                                                                                                                                
                      values );  
```
and the new `interpolatePoint` signature would be something like
```
interpolatePoint( IN_ARRAY const & axisCoordinates,                                                                                                                                    
                             arraySlice2d< real64 > const hypercubeData,                                                                                                                                  
                             real64 const * const axisLows,                                                                                                                                       
                             real64 const * const axisStepInvs,                                                                                                                                   
                             OUT_ARRAY && values ) const  
```
so I think there is a way to use LvArray instead of the raw pointers. However, let's do that in another PR, because this one is already quite large

Reply:
This is probably to deal with n-dimensional arrays dynamically.

Maybe there is a feature to tell `LvArray` that at this memory location lives data it does not own and that could be "served" as an `array3d`?